components/shared/ui/menu.tsx

```diff
@@ -1,53 +1,104 @@
 import Link from "next/link";
 import { FC, useState } from "react";
-import { mainColorBgClass } from "../../../lib/constants/colors";
+import { mainColorBgClass, mainColorBorderClass } from "../../../lib/constants/colors";
 import { useSiteCodename } from "../siteCodenameContext";
 import { Block_Navigation } from "../../../models";
 import { createItemSmartLink } from "../../../lib/utils/smartLinkUtils";
 import { ChevronDownIcon, Bars3Icon } from "@heroicons/react/24/solid";
+import { NextRouter, Router, useRouter } from "next/router";
 
 type Link = Readonly<Block_Navigation>;
 
 type Props = Readonly<{
   item: Link;
 }>
 
+type MenuListProps = Readonly<{
+  items: Block_Navigation[];
+  activeMenu: string | number;
+  smallMenuActive: boolean
+  handleClick: (menuId: string | number) => void;
+}>
+
 type DropdownMenuProps = Readonly<{
   links: ReadonlyArray<Link>;
 }>;
 
-const DropdownMenu: FC<DropdownMenuProps> = props => {
+const isCurrentNavigationItemActive = (navigation: Block_Navigation, router: NextRouter) => (
```

Comment:
Is there any reason to pass the whole router? I think that it can take path: string instead

Reply:
yes to keep the contract open, if I pass string, I can the just expect two strings and then just do the expression.